infrastructure/logging/src/main/java/tech/pegasys/teku/infrastructure/logging/StartupLogConfig.java

```diff
@@ -0,0 +1,183 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.infrastructure.logging;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.List;
+import oshi.hardware.HardwareAbstractionLayer;
+
+public class StartupLogConfig {
+  private final String network;
+  private final String storageMode;
+
+  private final String maxHeapSize;
+  private final String memory;
+  private final int cpuCores;
+
+  private final String beaconChainRestApiInterface;
+  private final int beaconChainRestApiPort;
+  private final List<String> beaconChainRestApiAllow;
+
+  private final String validatorRestApiInterface;
+  private final int validatorRestApiPort;
+  private final List<String> validatorRestApiAllow;
+
+  private final String executionEngineEndpoint;
+
+  public StartupLogConfig(
+      final String network,
+      final String storageMode,
+      final HardwareAbstractionLayer hardwareInfo,
+      final String beaconChainRestApiInterface,
+      final int beaconChainRestApiPort,
+      final List<String> beaconChainRestApiAllow,
+      final String validatorRestApiInterface,
+      final int validatorRestApiPort,
+      final List<String> validatorRestApiAllow,
+      final String executionEngineEndpoint) {
+    this.network = network;
+    this.storageMode = storageMode;
+
+    this.maxHeapSize = normalizeSize(Runtime.getRuntime().maxMemory());
+    this.memory = normalizeSize(hardwareInfo.getMemory().getTotal());
+    this.cpuCores = hardwareInfo.getProcessor().getLogicalProcessorCount();
+
+    this.beaconChainRestApiInterface = beaconChainRestApiInterface;
+    this.beaconChainRestApiPort = beaconChainRestApiPort;
+    this.beaconChainRestApiAllow = beaconChainRestApiAllow;
+
+    this.validatorRestApiInterface = validatorRestApiInterface;
+    this.validatorRestApiPort = validatorRestApiPort;
+    this.validatorRestApiAllow = validatorRestApiAllow;
+
+    this.executionEngineEndpoint = executionEngineEndpoint;
+  }
+
+  private String normalizeSize(final long size) {
+    return String.format("%.02f", (double) size / 1024 / 1024 / 1024) + " GB";
+  }
+
+  public List<String> getReport() {
+    final String general =
+        String.format("Configuration | Network: %s, Storage Mode: %s", network, storageMode);
+    final String host =
+        String.format(
+            "Host Configuration | Maximum Heap Size: %s, Total Memory: %s, CPU Cores: %d",
+            maxHeapSize, memory, cpuCores);
+    final String restApi =
+        String.format(
+            "Rest Api Configuration | Listen Address: %s, Port: %s, Allow: %s",
+            beaconChainRestApiInterface, beaconChainRestApiPort, beaconChainRestApiAllow);
```

Comment:
how does this go when rest api isn't available?

Reply:
Good point, I'll add a check for whether they are enabled. We shouldn't print that line if it is not.

Reply:
Actually it still comes out normal when rest api isn't enabled. Do we want that? Or should we be indicating it isn't enabled? Or removing the line?

Reply:
ok so it'll be grabbing defaults, so we want to probably add the enabled flag for those too, or not print them when enabled is false...